src/server/mod.rs

```diff
@@ -118,14 +118,13 @@ pub fn handle_client(mut stream: TcpStream,
 
                     // Send ServerHello message
                     let encoded: Vec<u8> = serialize(
-                        &Message::ServerHello(player_id, world.lock().unwrap().clone()),
+                        &Message::ServerHello(player_id, (*world).clone()),
```

Comment:
Style: I'd use `.deref()` to avoid the parentheses.

Reply:
Done.